### ISSUE TITLE
bug: Arguments missing in grid intersection with polygon 

### ISSUE THREAD

**Issue by borjav2** [modflowpy/flopy#1626]
Hi,

I searched for an issue similar but didn't find anything. The issue consists in that the intersect() method of a flopy.utils.gridintersect.GridIntersect class object doesn't allow "contains_centroid" and "min_area_threshold" arguments defined in the script. 

I noted this issue while running the "flopy3_grid_intersection_demo.ipynb" notebook which can be summarized below:

<details>
  <summary>Code</summary>

```
import os
import sys

import matplotlib as mpl
import matplotlib.pyplot as plt
import numpy as np

# run installed version of flopy or add local path
try:
    import flopy
    import flopy.discretization as fgrid
    import flopy.plot as fplot
    from flopy.utils.gridintersect import GridIntersect
except:
    fpth = os.path.abspath(os.path.join("..", ".."))
    sys.path.insert(1, fpth)
    import flopy
    import flopy.discretization as fgrid
    import flopy.plot as fplot
    from flopy.utils.gridintersect import GridIntersect

import shapely
from shapely.geometry import (LineString, MultiLineString, MultiPoint,
                              MultiPolygon, Point, Polygon)

print(sys.version)
print("numpy version: {}".format(np.__version__))
print("matplotlib version: {}".format(mpl.__version__))
print("flopy version: {}".format(flopy.__version__))

delc = 10 * np.ones(10, dtype=float)
delr = 10 * np.ones(10, dtype=float)
xoff = 0.0
yoff = 0.0
angrot = 0.0
sgr = fgrid.StructuredGrid(
    delc, delr, top=None, botm=None, xoff=xoff, yoff=yoff, angrot=angrot
)

p = Polygon(
    shell=[
        (15, 15),
        (20, 50),
        (35, 80.0),
        (80, 50),
        (80, 40),
        (40, 5),
        (15, 12),
    ],
    holes=[[(25, 25), (25, 45), (45, 45), (45, 25)]],
)

ix = GridIntersect(sgr, method="vertex", rtree=True)
result2 = ix.intersect(p, contains_centroid=True)
```

</details>

I would like to ask why those features were removed? Those were important features and I would be grateful if they were allowed again.

Cheers

Additional information:
 - OS:  Windows 10 22H2
 - Browser: Mozilla Firefox
 - FloPy: 3.3.5

**Comment by dbrakenhoff** [comment]
Hi @borjav2,

These features are currently only available in the develop branch and should be included in the next (3.3.6) release. So you could install the develop version and you should be good to go! Hope that helps?

**Comment by borjav2** [comment]
That works. Thanks a lot.
